pkg/kubernetes/watcher.go

```diff
@@ -151,17 +151,20 @@ func (w *ObjListWatcher) handleUpdate(obj interface{}) {
 			return
 		}
 		for _, condition := range pod.Status.Conditions {
-			if condition.Type != k8sv1.ContainersReady && condition.Status != k8sv1.ConditionTrue {
-				continue
-			}
-			w.Mx.Lock()
-			err1 := w.fillInfo(pod, pod.Status.ContainerStatuses)
-			err2 := w.fillInfo(pod, pod.Status.InitContainerStatuses)
-			err3 := w.fillInfo(pod, pod.Status.EphemeralContainerStatuses)
-			w.Mx.Unlock()
-			// only add pod to cache if all containers successfully added to map
-			if err1 == nil && err2 == nil && err3 == nil {
-				managedPods[podID] = true
+			klog.V(5).Infof("Pod %s %s status %s %s", pod.Name, pod.Namespace, condition.Type, condition.Status)
+			if condition.Type == k8sv1.ContainersReady {
```

Comment:
Is there any reason why you removed the `condition.Status != k8sv1.ConditionTrue`?

Reply:
when `ContainerReady` status is set, if the `Status` is `True`, then the Pods already initialized with ContainerIDs and `ContaienrMetrics` saves them. If the `Status` is `False`, then the `ContainerMetrics` is reset. 

But now I have another question, should we trust `managedPods` cache? What if the pod is updated with a new sidecar? The sidecar will not be checked since the podUID is already in the `managedPds` cache.

Reply:
Good point! Maybe we should not skip the events based on the managedPods....

Reply:
sounds good, i'll update

Reply:
@marceloamaral `managedPods` cache removed, all pods are to be evaluated.